File: android/src/main/java/com/contacttracing/ContactTracingModule.java
package com.contacttracing;

import androidx.annotation.Nullable;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.WritableMap;

import java.util.HashMap;
import java.util.Map;

public class ContactTracingModule extends ReactContextBaseJavaModule {
    private static final String E_ERROR = "E_ERROR";

    private final ReactApplicationContext reactContext;

    public ContactTracingModule(ReactApplicationContext reactContext) {
        super(reactContext);
        this.reactContext = reactContext;
    }

    @Override
    public String getName() {
        return "ContactTracing";
    }

    @Nullable
    @Override
    public Map<String, Object> getConstants() {
        HashMap<String, Object> constants = new HashMap<>();

        constants.put("supported", false);

        return constants;
    }

    @ReactMethod
    public void start(Promise promise) {
        promise.resolve("TODO");
    }

    @ReactMethod
    public void stop(Promise promise) {
        promise.resolve("TODO");
    }

    @ReactMethod
    public void currentStatus(Promise promise) {
        promise.resolve("TODO");
    }

    @ReactMethod
    public void requestExposureSummary(Promise promise) {
        promise.resolve("TODO");
    }
}

File: android/src/main/kotlin/com/contacttracing/ContactTracingModule.kt
package com.contacttracing

import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class ContactTracingModule(reactContext: ReactApplicationContext) :
  ReactContextBaseJavaModule(reactContext) {
  override fun getName() = "ContactTracing"

  override fun getConstants(): MutableMap<String, Any> {
    val constants = mutableMapOf<String, Any>()

    constants["supported"] = false

    return constants
  }

  @ReactMethod
  fun start(promise: Promise) {
    promise.resolve("TODO")
  }

  @ReactMethod
  fun stop(promise: Promise) {
    promise.resolve("TODO")
  }

  @ReactMethod
  fun currentStatus(promise: Promise) {
    promise.resolve("TODO")
  }

  @ReactMethod
  fun requestExposureSummary(promise: Promise) {
    promise.resolve("TODO")
  }
}